compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -476,22 +476,68 @@ static SmallVector<unsigned> padOperandsOfOp(RewriterBase &rewriter,
   return paddedOperands;
 }
 
+// Return a list of operands to be padded for each `op`.
+SmallVector<unsigned> shouldOperandsBePadded(Operation *op) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+  if (!linalgOp || !linalg::isaContractionOpInterface(linalgOp)) {
+    return {};
+  }
+
+  // Bail out on matvec / vecmat and skinny matmul problems.
+  SmallVector<unsigned> reductionDims;
+  linalgOp.getReductionDims(reductionDims);
+  int64_t parallelDimSize = 1;
+  llvm::SmallSetVector<int32_t, 4> reductionDimsSet;
+  reductionDimsSet.insert_range(reductionDims);
+  SmallVector<int64_t, 4> loopRanges = linalgOp.getStaticLoopRanges();
+  for (auto [idx, dimSize] : llvm::enumerate(loopRanges)) {
+    if (reductionDimsSet.contains(idx)) {
+      // Bail if the reduction dimension is dynamic.
+      if (ShapedType::isDynamic(dimSize)) {
+        return {};
+      }
+      continue;
+    }
+    if (!ShapedType::isDynamic(parallelDimSize)) {
+      if (ShapedType::isDynamic(dimSize)) {
+        parallelDimSize = ShapedType::kDynamic;
+        continue;
+      }
+      parallelDimSize *= dimSize;
+    }
+  }
+
+  // TODO(MaheshRavishankar): Make this command line controllable.
+  static constexpr int64_t kSkinnyMatmulThreshold = 64;
+  if (!ShapedType::isDynamic(parallelDimSize) &&
+      parallelDimSize < kSkinnyMatmulThreshold) {
+    // This matmul is skinny, do not pad.
+    return {};
+  }
```

Comment:
I agree with Max that the threshold seems to be target-specific. The TODO is dragging us to a worse state to me, because we are trying hard to move this kind of decision to codegen phase.

Reply:
Resolving this for now. I think this is a pretty light-weight decision, and potentially not something we are going to do by default any time soon. In final state we might not use this logic.